python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -867,6 +868,14 @@ def _config_type_for_type_on_pydantic_field(
     else:
         try:
             return convert_potential_field(potential_dagster_type).config_type
+        except DagsterInvalidDefinitionError:
+            raise DagsterInvalidPythonicConfigDefinitionError(
+                config_class=model_cls,
+                field_name=field_name,
+                invalid_type=potential_dagster_type,
+                is_resource=model_cls is not None
+                and safe_is_subclass(model_cls, ConfigurableResourceFactory),
+            )
```

Comment:
similar feedback as previous PR. We should explicilty check that this is instead the `model_cls` case rather than assuming the fallthrough is this case

Reply:
extracted check to just wrap annotation-checking segment of the code